src/test/compile-fail/fn-variance-1.rs

```diff
@@ -17,9 +17,9 @@ fn apply<T, F>(t: T, f: F) where F: FnOnce(T) {
 }
 
 fn main() {
-    apply(&3, takes_mut); //~ ERROR (values differ in mutability)
+    apply(&3, takes_mut); //~ ERROR type mismatch
```

Comment:
Yeah, that was a mistake. Apparently, that error doesn't pass through `span_err` and I never reverted it.